src/languages/nix.js

```diff
@@ -9,85 +9,291 @@ Category: system
-    ]
+      "toXML",
+      "trace",
+      "traceVerbose",
+      "true",
+      "tryEval",
+      "typeOf",
+      "unsafeDiscardOutputDependency",
+      "unsafeDiscardStringContext",
+      "unsafeGetAttrPos",
+      "warn",
+      "zipAttrsWith",
+    ].map(b => `builtins\\.${b}`).join('|') + ')'),
   };
-  const ANTIQUOTE = {
-    className: 'subst',
-    begin: /\$\{/,
-    end: /\}/,
-    keywords: KEYWORDS
+
+  const IDENTIFIER_REGEX = '[A-Za-z_][A-Za-z0-9_\'-]*';
+
+  const LOOKUP_PATH = {
+    className: 'symbol',
+    match: new RegExp(`<${IDENTIFIER_REGEX}(/${IDENTIFIER_REGEX})*>`),
   };
-  const ESCAPED_DOLLAR = {
-    className: 'char.escape',
-    begin: /''\$/,
+
+  const PATH = {
+    className: 'symbol',
+    match: new RegExp(`(\\.\\.|\\.|~)?/(${IDENTIFIER_REGEX})?(/${IDENTIFIER_REGEX})*`),
+  };
+
+  const OPERATOR = {
+    className: 'operator',
+    relevance: 0,
+    match: /(==|=|\+\+|\+|<=|<\||<|>=|>|->|-|\/\/|\/|!=|!|\|\||\|>|\?|\*|&&)/,
   };
+
   const ATTRS = {
-    begin: /[a-zA-Z0-9-_]+(\s*=)/,
+    begin: new RegExp(`${IDENTIFIER_REGEX}(\\s*=)`),
     returnBegin: true,
     relevance: 0,
     contains: [
       {
         className: 'attr',
         begin: /\S+/,
         relevance: 0.2
-      }
-    ]
+      },
+    ],
+  };
+
+  const NORMAL_ESCAPED_DOLLAR = {
+    className: 'char.escape',
+    match: /\\\$/,
+  };
+  const INDENTED_ESCAPED_DOLLAR = {
+    className: 'char.escape',
+    match: /''\$/,
+  };
+  const ANTIQUOTE = {
+    className: 'subst',
+    begin: /\$\{/,
+    end: /\}/,
+    keywords: KEYWORDS,
+  };
+  const ESCAPED_DOUBLEQUOTE = {
+    className: 'char.escape',
+    match: /'''/,
+  };
+  const ESCAPED_LITERAL = {
+    className: 'char.escape',
+    match: /\\./,
   };
   const STRING = {
     className: 'string',
-    contains: [ ESCAPED_DOLLAR, ANTIQUOTE ],
     variants: [
       {
         begin: "''",
-        end: "''"
+        end: "''",
+        contains: [
+          INDENTED_ESCAPED_DOLLAR,
+          ANTIQUOTE,
+          ESCAPED_DOUBLEQUOTE,
+          ESCAPED_LITERAL,
+        ],
       },
       {
         begin: '"',
-        end: '"'
-      }
-    ]
+        end: '"',
+        contains: [
+          NORMAL_ESCAPED_DOLLAR,
+          ANTIQUOTE,
+          ESCAPED_LITERAL,
+        ],
+      },
+    ],
   };
+
+  const URL = {
+    className: 'link',
+    match: RegExp("https?://[A-Za-z0-9\\-._~:/?#[\\]@!$&'()*+,=]+"),
+  };
+
   const EXPRESSIONS = [
     hljs.NUMBER_MODE,
     hljs.HASH_COMMENT_MODE,
     hljs.C_BLOCK_COMMENT_MODE,
+    hljs.COMMENT(
+      /\/\*\*(?!\/)/,
+      /\*\//,
+      {
+        subLanguage: 'markdown',
+        relevance: 0
+      }
+    ),
+    BUILTINS,
     STRING,
-    ATTRS
+    URL,
+    LOOKUP_PATH,
+    PATH,
+    ATTRS,
+    OPERATOR,
   ];
+
   ANTIQUOTE.contains = EXPRESSIONS;
+
+  const REPL = [
+    {
+      className: 'meta.prompt',
+      match: /nix-repl> /
+    },
+    {
+      className: 'meta',
+      match: /:[a-z]+/,
```

Comment:
Can we pin to the beginning of line here perhaps with `^`... not sure, just a suggestion.

Reply:
I have pinned `nix-repl>` above with a `^`. `:[a-z]+` is meant to highlight commands used in the REPL after the prompt. There isn't anything else that can come after a colon in the nix language for now, so I'm not sure how much better we could make this one.